deepspeed/comm/comm.py

```diff
@@ -189,26 +186,34 @@ def is_available() -> bool:
     return True
 
 
-def set_backend(backend_name):
-    if not use_ds_backend:
-        utils.logger.error(
-            "DeepSpeed communication backend is required. Please use deepspeed.comm.init_distributed(backend, use_deepspeed=True) to use this functionality"
-        )
-        raise RuntimeError('Error: Custom DeepSpeed backend called without initializing DeepSpeed distributed.')
+def set_backend(timeout=None, init_method=None):
+    # if user specific --prefer_deepspeed_comm when launching deepspeed, will load cdb to
+    # DeepSpeed backend, otherwise fallback to torch distributed backend
 
     global cdb
     global nccl_backend
     global mpi_backend
-
-    try:
-        if backend_name == NCCL_BACKEND:
-            if nccl_backend is not None and nccl_backend.is_initialized():
-                cdb = nccl_backend
-        elif backend_name == MPI_BACKEND:
-            if mpi_backend is not None and mpi_backend.is_initialized():
-                cdb = mpi_backend
-    except Exception as inst:
-        print(inst)
+    global ccl_backend
+
+    backend_name = get_accelerator().communication_backend_name()
+
+    if backend_name == NCCL_BACKEND:
+        if nccl_backend is not None and nccl_backend.is_initialized():
+            cdb = nccl_backend
+    elif backend_name == MPI_BACKEND:
+        if mpi_backend is not None and mpi_backend.is_initialized():
+            cdb = mpi_backend
+    elif backend_name == CCL_BACKEND:
+        if ccl_backend is None:
```

Comment:
Why ccl not be handled like other backends and code simplified to 

```python
try: 
  ...
  elif backend_name == CCL_BACKEND:
    if ccl_backend is not None and ccl_backend .is_initialized():
      cdb = ccl_backend 
except ...
```

Reply:
> Why ccl not be handled like other backends and code simplified to
> 
> ```python
> try: 
>   ...
>   elif backend_name == CCL_BACKEND:
>     if ccl_backend is not None and ccl_backend .is_initialized():
>       cdb = ccl_backend 
> except ...
> ```

Yes, this part should be simplified and backend creation should be put into init_deepspeed_backend.  The latest comm code reflect this change.

Reply:
Thanks. With that change, I think `--prefer_deepspeed_comm`  is no longer needed and can be removed.

Reply:
Will remove prefer_deepspeed_comm and use CCLBackend as default CPU backend.  Note that CCLBackend only implemented allreduce/broadcast/barrier so it works for inference only.   We could extend it to other collectives if training support is needed.

Reply:
Yes, I am aligned with CCLBackend as default CPU backend